xscen/aggregate.py

```diff
@@ -341,32 +342,75 @@ def spatial_mean(
 
     """
 
+    def get_spatial_dims(ds):
+        # Determine the X and Y names
+        spatial_dims = {}
+        for d in ["X", "Y"]:
+            if d in ds.cf.axes:
+                spatial_dims[d] = ds.cf[d].name
+            elif (
+                (d == "X")
+                and ("longitude" in ds.cf.coordinates)
+                and (len(ds[ds.cf.coordinates["longitude"][0]].dims) == 1)
+            ):
+                spatial_dims[d] = ds.cf.coordinates["longitude"]
+            elif (
+                (d == "Y")
+                and ("latitude" in ds.cf.coordinates)
+                and (len(ds[ds.cf.coordinates["latitude"][0]].dims) == 1)
+            ):
+                spatial_dims[d] = ds.cf.coordinates["latitude"]
+
+        return spatial_dims
+
     kwargs = kwargs or {}
 
     # If requested, call xscen.extraction.clisops_subset prior to averaging
     if call_clisops:
         ds = clisops_subset(ds, region)
 
+    if method == "cos-lat":
+        # Determine the latitude name
+        spatial_dims = get_spatial_dims(ds)
+        if ("latitude" in ds.cf.coordinates) and (
+            len(ds[ds.cf.coordinates["latitude"][0]].dims) > 1
+        ):
+            raise NotImplementedError(
+                "cos-lat averaging is not yet supported for 2D grids."
+            )
+        if spatial_dims.get("Y") is None:
+            raise ValueError(
+                "Could not determine the latitude name using CF conventions. "
+                "Use kwargs = {lat: str} to specify on which dimension to perform the averaging."
+            )
+
+        if "units" not in ds[spatial_dims["Y"]].attrs:
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude does not appear to have units. Make sure that the computation is right."
+            )
+        elif ds[spatial_dims["Y"]].attrs["units"] != "degrees_north":
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude units is '{ds[spatial_dims['Y']].attrs['units']}', expected 'degrees_north'. "
+                f"Make sure that the computation is right."
+            )
+
+        weight = np.cos(np.deg2rad(ds[spatial_dims["Y"]]))
+        weight = weight / weight.sum()
+        ds_agg = ds.weighted(weight).mean(
+            [d for d in spatial_dims.values()], keep_attrs=True
+        )
+
```

Comment:
I'm not sure I understand why you are doing this over the "Y" dimension and not over "latitude" directly. Like, why is 
```
weight = np.cos(np.deg2rad(ds.cf['latitude']))
weight = weight / weight.sum()
```
not correct?

AFAIU, this also support 2D grids.

You need `spatial_dims` for the effective `mean` call, but `spatial_dims["Y"]` is `rlat` in rotated pole, which is NOT wanted for the `cos-lat` method, no?

Reply:
Yes, we should be using 'lat' rather than 'rlat' for the weights in these cases, and I can confirm that using ds.cf['latitude'] (rather than spatial_dims['Y']) does work with 2D rotated grids. I tested this with a CRCM5 monthly statistics file with spatial_dims rlat and rlon, and the following works properly: 
```
weight = np.cos(np.deg2rad(ds.cf['latitude'])) 
da_weighted = ds['tas_moy'].weighted(weight).mean([d for d in spatial_dims.values()])
```